Support for multi room providers (#1226)
liveblocks/liveblocks: docs/pages/api-reference/liveblocks-client.mdx

  mutable by every client. Must either contain Live structures (e.g.
  `new LiveList()`, `new LiveObject({ a: 1 })`, etc.) or be serializable to
  JSON.
- `autoConnect` (optional) - Whether or not the room immediately connects to
  Liveblocks servers. Default is `true`. Usually set to `false` when the client
  is used from the server to not call the authentication endpoint or connect via
  WebSocket.

```ts
const { room, leave } = client.enterRoom("my-room", {
  initialPresence: { cursor: null },
  initialStorage: { todos: new LiveList() },
});

// Don’t forget to call `leave()` when you’re done with the room
leave();
```

### Client.enter

Enters a room and returns its local `Room` instance.

<Banner title="Deprecated since 1.5" type="warning">

> We no longer recommend using this API, because it will cause issues if you have
> multiple component trees in your application that connect to the same room.
> Prefer the new and improved [`client.enterRoom`][] API.

Reviewer: ```suggestion
We no longer recommend this API, because issues will occur if you 
simultaneously connect to the same room with multiple instances.
We recommend using [`client.enterRoom`][] instead.
```

Wow, this is really hard to word! Does this work?

Please update this elsewhere on the page!

Author: Thanks, this is much better! 👍

const status = room.getStorageStatus();
```

### Room.connect

Connect the local room instance to the Liveblocks server. Does nothing if the
room is already connecting, reconnecting or connected.

```ts
room.connect();
```

We don’t recommend using this API directly.

### Room.reconnect

Reconnect the local room instance to the Liveblocks server, using a new
WebSocket connection.

```ts
room.reconnect();
```

### Room.disconnect

> Disconnect the local room instance from the Liveblocks server. The room instance
> will remain functional (for example, it will still allow local presence or
> storage mutations), but since there is no connectivity any longer, those changed
> will not be persisted or synchronized until the room instance is reconnected to
> the server again.

Reviewer: ```suggestion
Disconnect the local room instance from the Liveblocks server. The room instance
will remain functional (for example, it will still allow local presence or
storage mutations), but since it’s no longer connected, changes
will not be persisted or synchronized until the room instance is reconnected again.
```

Author: Love it!
